jax/interpreters/xla.py

```diff
@@ -347,7 +347,13 @@ def write(v, node):
                  map(aval, eqn.invars), backend, *in_nodes, **new_params)
     elif eqn.primitive in parallel_translations:
       replica_groups = axis_groups(axis_env, eqn.params['axis_name'])
-      new_params = {k: v for k, v in eqn.params.items() if k != 'axis_name'}
+      axis_index_groups = eqn.params.get('axis_index_groups', None)
```

Comment:
We shouldn't mutate `eqn.params`, and also it's a (related) smell to need a default here. We should make sure `axis_index_groups`, is _always_ populated from the start.

To be concrete: change this to `params['axis_index_groups']`?

Reply:
I don't think `d.get(key, default)` mutates `d` (just checked).
Also, the reason `axis_index_groups` isn't always populated is because it's not present on all parallel primitives, just the ones that lower to `AllReduce`.  I believe it doesn't really make sense for `ppermute`, and while XLA `AllToAll` has a `replica_groups` attribute I'm not quite sure how we should expose that in the relevant lax wrappers.

Reply:
Total brain fart on my part. I saw another `params.pop` elsewhere (where it was safe) and confused it with this one. Sorry for the noise!

Thanks for explaining the fact that it's not on all parallel primitives. That makes sense. I withdraw my suggestions!